### ISSUE TITLE
Run accessibility audit on CI

### ISSUE THREAD

**Issue by jelbourn** [angular/material2#262]
Tools to explore:
Chrome tools: https://github.com/GoogleChrome/accessibility-developer-tools
aXe: http://www.deque.com/products/axe/

The audit should run on the e2e examples.

**Comment by sendilkumarn** [comment]
We can use  [Protractor-accessibility-plugin](https://github.com/angular/protractor-accessibility-plugin) ? 

adding this to protractor.conf.js 

<img width="915" alt="screen shot 2016-05-25 at 12 52 51 pm" src="https://cloud.githubusercontent.com/assets/12471122/15528576/a037baa0-2277-11e6-8c7f-a411f4f2a75f.png">

**Comment by sendilkumarn** [comment]
if this is okay. I will do a PR for this :) @jelbourn

**Comment by DevVersion** [comment]
I think that makes sense. I'm not in position to confirm this, but I think a PR is always welcome.

btw: I'm working a PR which uses Axe Core. So our work will not interfere :)

**Comment by sendilkumarn** [comment]
Okay then 👍

**Comment by jelbourn** [comment]
In the long term we probably just want to go with one a11y assertion system, but it could be good to compare them initially.

**Comment by DevVersion** [comment]
@jelbourn I think both a11y assertion systems are powerful.

When I see the simplicity of the protractor plugin, I think this one is way easier to handle.

I already got Axe Core running on the e2e app as well.
Those A11Y checks can be triggered / started manually by adding a line to the given jasmine spec.

To get Axe Core running, we have to
- Copy it to the vendors for e2e and load it in the e2e
 / Or inject it manually on the given spec (recommended)

- Manually log our violations to the console or file (really difficult to show all necessary information)

Maybe @marcysutton can help in this case, and explain whats different about axe :)

**Comment by marcysutton** [comment]
The Protractor plugin will only run the A11y checks once at the end of a test and it won't let you specify an HTML fragment to test against (it tests the whole page). aXe, on the other hand, will allow you to run checks whenever the UI has changed within a set of tests and even test a DOM subtree. 

The Chrome A11y Developer tools are also a smaller, noisier ruleset than aXe, and Tenon requires an API key and paid account.

Here are a few questions you should consider: 

- What should be done with the log output? (We could easily write a utility for this)
- How will developers respond to accessibility issues? (for example, would developers run the tests and look at the debugger?)
- Would the issues go into an issue tracker?
- If you are analyzing whole pages, how do you plan to de-dupe the issues that are common from test-to-test? (None of the mentioned tools do this, but it's worth thinking about.)

**Comment by jelbourn** [comment]
I'd actually like to have tests _fail_ when the a11y audit encounters an issue.

**Comment by marcysutton** [comment]
You can easily do that, it just might take some more logging to make the failure understandable. Here is an example: https://github.com/marcysutton/axe-webdriverjs-demo/blob/master/spec/test.js#L54

**Comment by jelbourn** [comment]
Yeah, that seems like a good area for some custom jasmine matchers.

**Comment by DevVersion** [comment]
@jelbourn I already got that working without the custom webdriver of the axe-core library.

It as bit messy at the moment, because it's just a WIP branch (on vacation) 
- https://github.com/DevVersion/material2/commit/d2af0f85dc37606e8939fb800f496c54b99f239c

I'm looking for some good designs, to represent the violations. 
> Colors would be appropriated, but will require libraries like `chalk`

**Comment by sendilkumarn** [comment]
Protractor tests fail when there are any accessibility issues.

**Comment by DevVersion** [comment]
Closing this, because #2010 just landed in `master`.
